auth-providers-common/pkg/state/state.go

```diff
@@ -40,40 +40,45 @@ func ObotGetState(p *oauth2proxy.OAuthProxy) http.HandlerFunc {
 
 		reqObj.Header = sr.Header
 
-		state, err := p.LoadCookiedSession(reqObj)
+		ss, err := GetSerializableState(p, reqObj)
 		if err != nil {
-			http.Error(w, fmt.Sprintf("failed to load cookied session: %v", err), http.StatusBadRequest)
+			http.Error(w, fmt.Sprintf("failed to get state: %v", err), http.StatusInternalServerError)
 			return
 		}
 
-		if state == nil {
-			http.Error(w, "state is nil", http.StatusInternalServerError)
+		if err = json.NewEncoder(w).Encode(ss); err != nil {
+			http.Error(w, fmt.Sprintf("failed to encode state: %v", err), http.StatusInternalServerError)
 			return
 		}
+	}
+}
 
-		var setCookies []string
-		if state.IsExpired() || (p.CookieOptions.Refresh != 0 && state.Age() > p.CookieOptions.Refresh) {
-			setCookies, err = refreshToken(p, reqObj)
-			if err != nil {
-				http.Error(w, fmt.Sprintf("failed to refresh token: %v", err), http.StatusForbidden)
-				return
-			}
-		}
+func GetSerializableState(p *oauth2proxy.OAuthProxy, r *http.Request) (SerializableState, error) {
```

Comment:
This is a refactoring of some of the `ObotGetState` behavior so that I could call just this function to get the SerializableState, and then modify it afterwards. We needed this for GitHub.